internal/federation/handle.go

```diff
@@ -18,7 +18,7 @@ import (
 func MakeJoinRequestsHandler(s *Server, w http.ResponseWriter, req *http.Request) {
 	// Check federation signature
 	fedReq, errResp := gomatrixserverlib.VerifyHTTPRequest(
-		req, time.Now(), gomatrixserverlib.ServerName(s.ServerName), s.keyRing,
+		req, time.Now(), gomatrixserverlib.ServerName(s.serverName), s.keyRing,
```

Comment:
I haven't gone through these in detail, but it seems problematic that they are changing to `serverName` rather than `ServerName()`.

AIUI, the difference is that `serverName` works before we start listening, but *may return an incomplete server name*. If we pass the wrong server name in here, aren't we checking that the request is signed for the wrong server?

Reply:
No because this is called when the server is already running, hence `s.serverName` is already set to the right value.

Reply:
In which case, why not call `ServerName()`, to include the safety check?

Reply:
Because there is no need for the safety check. The server _must_ be listening at this point in order for this code to be hit.

Reply:
for now, sure. But this is about robustness, and failing in a clear way when an assumption is violated.

Anyway, I don't feel that strongly. If you don't think the check is worthwhile, feel free to skip it. I'm not going to go through each of these changes to check that .listen() must have happened first though, so I'll have to take your word for it.